plugins/casutg/src/main/java/com/blazemeter/jmeter/threads/concurrency/ConcurrencyThreadStarter.java

```diff
@@ -53,14 +70,32 @@ private long getPlannedConcurrency() {
         }
 
         if (rampUp == 0 || timeOffset > rampUp) {
-            return Math.round(maxConcurr);
+            return maxConcurr;
         } else if (steps > 0) {
             double stepSize = maxConcurr / (double) steps;
             double stepLen = rampUp / (double) steps;
             return Math.round(stepSize * (Math.floor(timeOffset / stepLen) + 1));
         } else {
-            double slope = maxConcurr / rampUp;
+            double slope = (double) maxConcurr / rampUp;
             return Math.round(slope * timeOffset);
         }
     }
+
+    void checkNeedsPropertiesReloading(long now) {
+        if (CACHING_VALIDITY_MS > 0 && now - lastCachedTime > CACHING_VALIDITY_MS) {
+            this.rampUp = owner.getRampUpSeconds();
+            this.hold = owner.getHoldSeconds();
+            this.steps = owner.getStepsAsLong();
+            this.maxConcurr = Math.round(owner.getTargetLevelAsDouble());
+            this.lastCachedTime = System.currentTimeMillis();            
+        }
+    }
+
+    /**
+     * @return the lastCachedTime
+     */
+    long getLastCachedTime() {
```

Comment:
If this is for testing needs, it should have `protected` access and be exposed via `ConcurrencyThreadGroupExt` class

Reply:
or have `lastCachedTime` as protected and move this method to `ConcurrencyThreadGroupExt`

Reply:
I don't understand, ConcurrencyThreadGroupExt does not extend ConcurrencyThreadStarter.
Can you clarify ? 
Thanks

Reply:
Probably I'm wrong and misunderstood the change. My bad